kie-server-parent/kie-server-services/kie-server-services-prometheus/src/main/java/org/kie/server/services/prometheus/PrometheusProcessEventListener.java

```diff
@@ -0,0 +1,126 @@
+            .help("Kie Server Completed Process Instances")
+            .labelNames("container_id", "process_id", "status")
+            .register();
+
+    protected static final Gauge runningProcessInstances = Gauge.build()
+            .name("kie_server_process_instance_running_total")
+            .help("Kie Server Running Process Instances")
+            .labelNames("container_id", "process_id")
+            .register();
+
+    protected static final Summary processInstancesDuration = Summary.build()
+            .name("kie_server_process_instance_duration_seconds")
+            .help("Kie Server Process Instances Duration")
+            .labelNames("container_id", "process_id")
+            .register();
+
+    protected static final Summary workItemsDuration = Summary.build()
+            .name("kie_server_work_item_duration_seconds")
+            .help("Kie Server Work Items Duration")
+            .labelNames("name")
+            .register();
+
+    protected static void recordRunningProcessInstance(String containerId, String processId) {
+        runningProcessInstances.labels(containerId, processId).inc();
+    }
+
+    @Override
+    public void afterProcessStarted(ProcessStartedEvent event) {
+        LOGGER.debug("After process started event: {}", event);
+        final WorkflowProcessInstanceImpl processInstance = (WorkflowProcessInstanceImpl) event.getProcessInstance();
+        numberOfProcessInstancesStarted.labels(processInstance.getDeploymentId(), processInstance.getProcessId()).inc();
+        recordRunningProcessInstance(processInstance.getDeploymentId(), processInstance.getProcessId());
+    }
+
+    @Override
+    public void afterProcessCompleted(ProcessCompletedEvent event) {
+        LOGGER.debug("After process completed event: {}", event);
+        final WorkflowProcessInstanceImpl processInstance = (WorkflowProcessInstanceImpl) event.getProcessInstance();
+        runningProcessInstances.labels(processInstance.getDeploymentId(), processInstance.getProcessId()).dec();
+
+        numberOfProcessInstancesCompleted.labels(processInstance.getDeploymentId(), processInstance.getProcessId(), valueOf(processInstance.getState())).inc();
+
+        if (processInstance.getStartDate() != null) {
+            final double duration = millisToSeconds(System.currentTimeMillis() - processInstance.getStartDate().getTime());
+            processInstancesDuration.labels(processInstance.getDeploymentId(), processInstance.getProcessId()).observe(duration);
+            LOGGER.debug("Process Instance duration: {}s", duration);
+        }
+    }
+
+    @Override
+    public void beforeNodeLeft(ProcessNodeLeftEvent event) {
+        LOGGER.debug("Before Node left event: {}", event);
+        final NodeInstance nodeInstance = event.getNodeInstance();
+        if (nodeInstance instanceof WorkItemNodeInstance) {
+            WorkItemNodeInstance wi = (WorkItemNodeInstance) nodeInstance;
+            if (wi.getTriggerTime() != null) {
```

Comment:
Probably the last thing I noticed. Can this even be null? According to this I don't think so:
https://github.com/kiegroup/jbpm/pull/1473/files#diff-0d58c1b1aa778882879a180d7b1f90bdR386
(if you think it is safer to check it first, ignore this comment)